config/config.go

```diff
@@ -0,0 +1,169 @@
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+type LookupFn func(key string) (string, bool)
+
+type Config struct {
+	Workspace *workspace               `toml:"workspace"`
+	Analysis  map[string]*AnalysisConf `toml:"analysis"`
+}
+
+type workspace struct {
+	Version    string   `toml:"version"`
+	RootDir    string   `toml:"root"`
+	PolicyDir  string   `toml:"policies"`
+	IgnoreList []string `toml:"ignore-list"`
+	GoArch     string   `toml:"go-arch"`
+}
+
+type AnalysisConf struct {
+	Level  string
+	Target string
+}
+
+type ConfOptFn func(*Config)
+
+func NewConfig(options ...ConfOptFn) *Config {
+	conf := &Config{
+		Workspace: &workspace{
+			Version:    defaultVersion,
+			RootDir:    defaultRoot,
+			PolicyDir:  defaultPolicies,
+			IgnoreList: defaultIgnoredList,
+			GoArch:     defaultGoArch,
+		},
+	}
+
+	for i := 0; i < len(options); i++ {
+		options[i](conf)
+	}
+
+	return conf
+}
+
+func WithRootDir(dirPath string) ConfOptFn {
+	return func(c *Config) {
+		c.Workspace.RootDir = dirPath
+	}
+}
+
+func WithPolicyDir(dirPath string) ConfOptFn {
+	return func(c *Config) {
+		c.Workspace.PolicyDir = dirPath
+	}
+}
+
+func WithIgnoreList(ignoreList []string) ConfOptFn {
+	return func(c *Config) {
+		c.Workspace.IgnoreList = ignoreList
+	}
+}
+
+func WithGoArch(goArch string) ConfOptFn {
+	return func(c *Config) {
+		c.Workspace.GoArch = goArch
+	}
+}
+
+func WithAnalysis(analysis map[string]*AnalysisConf) ConfOptFn {
+	return func(c *Config) {
+		c.Analysis = analysis
+	}
+}
+
+type ConfReadFileOptFn func(*ReadConf)
+
+type ReadConf struct {
+	lookup      func(key string) (string, bool)
+	readFile    func(name string) ([]byte, error)
+	readToml    func(data string, v interface{}) (toml.MetaData, error)
+	interpolate func(lookup LookupFn, strWithEnvs string) (string, error)
+}
+
+func NewFromFileOpt(options ...ConfReadFileOptFn) *ReadConf {
+	readConf := &ReadConf{
+		lookup:      os.LookupEnv,
+		readFile:    os.ReadFile,
+		readToml:    toml.Decode,
+		interpolate: interpolate,
+	}
+
+	for i := 0; i < len(options); i++ {
+		options[i](readConf)
+	}
+
+	return readConf
+}
+
+func NewConfigFromFile(filePath string, options ...ConfReadFileOptFn) (*Config, error) {
+	readConf := NewFromFileOpt(options...)
+
+	conf := NewConfig()
+
+	data, err := readConf.readFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	dataWithEnv, err := readConf.interpolate(readConf.lookup, string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := readConf.readToml(dataWithEnv, &conf); err != nil {
+		return nil, err
+	}
+
+	return conf, nil
+}
+
+func (cfg *Config) Validate() error {
+	// if cfg.Version == defaultVersion {
+	// 	return errors.New("version field must be set")
+	// }
+
+	// if _, versionAvailable := availableVersions[cfg.Version]; !versionAvailable {
+	// 	return errors.New("specified version is not available")
+	// }
+
+	// ToDo: implement validate function
+
+	return nil
+}
```

Comment:
commented out the code. IF the func is not implemented yet, just put it in to a ToDo list somwhere in to the top of the file